File: core/src/no/elg/infiniteBootleg/console/CmdArgNames.java
package no.elg.infiniteBootleg.console;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface CmdArgNames {

    /**
     * @return Names of arguments to the annotated command (in order)
     */
    String[] value();
}

File: core/src/main/kotlin/no/elg/infiniteBootleg/console/CmdArgNames.kt
package no.elg.infiniteBootleg.console

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class CmdArgNames(
  /**
   * @return Names of arguments to the annotated command (in order)
   */
  vararg val value: String
)
